app/services/form_response.rb

```diff
@@ -15,6 +15,16 @@ def to_h
     { success: success, errors: errors }.merge!(extra)
   end
 
+  def merge(other)
+    errors = @errors.presence || other.errors
+    errors = Hash[errors.collect { |item| [item, ''] }] if errors.class == Array
```

Comment:
This doesn't combine the two errors? It just picks one? My expectation is that we'd find a way to include items from both errors?

Maybe if adding an array to a hash, we could do:

```ruby
if other.errors.kind_of?(Array)
  @errors.merge(other: other.errors) # add a new key with array of errors
else
  @errors.merge(other.errors)
end
```

Reply:
There can never be two errors.  Once there is one response with an error it short circuits

Reply:
Essentially you will keep adding successes until the last success or failure.

Reply:
That seems inflexible? I think there could be two error responses that combine

Reply:
Also since this is a new method for FormResponse, can we get specs for it?

Reply:
Also for the conversion of an array into a hash, many of the values in our error hashes are already arrays themselves, WDYT of this shape?

```ruby
if other.errors.is_a?(Array)
  { other: other.errors }
end
```

Reply:
Also I checked with @jmhooper and I'm going to see if I can refactor the errors on DocAuth::Response to be a hash, it'll match our FormResponse better

Reply:
I like it.  One of my PRs had it similar to that

Reply:
Yeah, I'd love to see the errors from both responses combined, that way if we ever do try to chain them they behave the way we'd expect.

I opened [this ticket](https://cm-jira.usa.gov/browse/LG-3359) to convert `#errors` on `DocAuth::Response` to return a hash. That way this is as easy deep merging hashing.